core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptorStorage.java

```diff
@@ -158,6 +161,7 @@ private synchronized void updateMemoryReservation(long delta)
                     .orElseThrow(() -> new VerifyException(format("storage is empty but reservedBytes (%s) is still greater than maxMemoryInBytes (%s)", reservedBytes, maxMemoryInBytes)));
             TaskDescriptors storage = storages.get(killCandidate);
             long previousReservedBytes = storage.getReservedBytes();
+            log.info("Failing query %s; reclaiming %s of %s task descriptor memory from %s queries", killCandidate, storage.getReservedBytes(), succinctBytes(reservedBytes), storages.size());
```

Comment:
I wonder if it is worth adding more details here (like for example an average task descriptor size / average split size / etc.)

Reply:
Added